tools/lkl/tests/net-test.c

```diff
@@ -44,41 +77,59 @@ in_cksum(const u_short *addr, register int len, u_short csum)
 	return answer;
 }
 
-static char *dst;
-static int test_icmp(char *str, int len)
+static int lkl_test_sleep(void)
+{
+	struct lkl_timespec ts = {
+		.tv_sec = cla.sleep,
+	};
+	int ret;
+
+	ret = lkl_sys_nanosleep(&ts, NULL);
+	if (ret < 0) {
+		lkl_test_logf("nanosleep error: %s\n", lkl_strerror(ret));
+		return TEST_FAILURE;
+	}
+
+	return TEST_SUCCESS;
+}
+
+static int lkl_test_icmp(void)
 {
 	int sock, ret;
-	struct iphdr *iph;
-	struct icmphdr *icmp;
+	struct lkl_iphdr *iph;
+	struct lkl_icmphdr *icmp;
 	struct sockaddr_in saddr;
 	struct lkl_pollfd pfd;
 	char buf[32];
 
-	str += snprintf(str, len, "%s ", dst);
+	if (cla.dst == INADDR_NONE)
+		return TEST_SKIP;
+
+	memset(&saddr, 0, sizeof(saddr));
+	saddr.sin_family = AF_INET;
+	saddr.sin_addr.s_addr = cla.dst;
 
-	sock = lkl_sys_socket(AF_INET, SOCK_RAW, IPPROTO_ICMP);
+	lkl_test_logf("pinging %s\n", inet_ntoa(saddr.sin_addr));
+
+	sock = lkl_sys_socket(LKL_AF_INET, LKL_SOCK_RAW, LKL_IPPROTO_ICMP);
 	if (sock < 0) {
-		snprintf(str, len, "socket error (%s)", lkl_strerror(sock));
+		lkl_test_logf("socket error (%s)\n", lkl_strerror(sock));
 		return TEST_FAILURE;
 	}
 
 	icmp = malloc(sizeof(struct icmphdr *));
```

Comment:
maybe it's okay but `sizeof(struct lkl_icmphdr *)` instead ?

Reply:
Yes, good catch, I'll fix it.